great_expectations/cli/toolkit.py

```diff
@@ -406,9 +417,20 @@ def load_data_context_with_error_handling(
     directory: str, from_cli_upgrade_command: bool = False
 ) -> DataContext:
     """Return a DataContext with good error handling and exit codes."""
-    # TODO consolidate all the myriad CLI tests into this
     try:
-        context = DataContext(directory)
+        context: Optional[DataContext] = DataContext(directory)
+        ge_config_version: int = context.get_config().config_version
+        if int(ge_config_version) >= GE_CONFIG_VERSION_WITH_EXPERIMENTAL_CHECKPOINTS:
+            (
+                increment_version,
+                exception_occurred,
+            ) = upgrade_project_one_version_increment(
+                context_root_dir=directory,
+                ge_config_version=ge_config_version,
+                continuation_message=EXIT_UPGRADE_CONTINUATION_MESSAGE,
+            )
```

Comment:
I think there should be a constant for the `GE_VERSION_WITH_EXPERIMENTAL_CHECKPOINTS` (i.e. 0.13), and the check here should be 

```
if (int(ge_config_version) < GE_CONFIG_VERSION_WITH_EXPERIMENTAL_CHECKPOINTS
    and current_ge_version >= GE_VERSION_WITH_EXPERIMENTAL_CHECKPOINTS
)
```
Which means `GE_CONFIG_VERSION_WITH_EXPERIMENTAL_CHECKPOINTS` should be `3`, not `2`.

Reply:
This seems to me the other way around.  The GE config with "real" checkpoints is 3.  The GE config with experimental checkpoints is 2.  So the value of the constant `GE_CONFIG_VERSION_WITH_EXPERIMENTAL_CHECKPOINTS` is 2.

Reply:
Oh, I see. Then in my comment above, GE_CONFIG_VERSION_WITH_EXPERIMENTAL_CHECKPOINTS should have been GE_CONFIG_VERSION_WITH_CHECKPOINTS (or whatever name). I suppose it feels weird triggering `upgrade_project_one_version_increment` based only on a config being version 2 or above (particularly if it's already version 3). Don't we just want to trigger `upgrade_project_one_version_increment` if we see something < 3 while someone is on 0.13 or above?

Reply:
The problem there is that even if instantiating DataContext does not trigger an exception, we still want to run an upgrade, because we might be on version 2.  I am thinking of how to fix it cleanly.